lib/srv/db/proxyserver.go

```diff
@@ -119,7 +119,8 @@ func (s *ProxyServer) Serve(listener net.Listener) error {
 		// The connection is expected to come through via multiplexer.
 		clientConn, err := listener.Accept()
 		if err != nil {
-			if strings.Contains(err.Error(), teleport.UseOfClosedNetworkConnection) {
+			if strings.Contains(err.Error(), teleport.UseOfClosedNetworkConnection) ||
+				strings.Contains(err.Error(), teleport.ListenerIsClosed) {
```

Comment:
seems hacky, I would go the other way - why not create an error compatible type for this problem and use `trace.IsAlreadyClosed` error here, instead of using string submatches

Reply:
@klizhentas Good point, seems like `trace.ConnectionProblem` should cover this as this is what multiplexer returns so no new error type seems to be needed. Updated.